src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/Ssse3Helper.cs

```diff
@@ -0,0 +1,103 @@
+            // Therefore we split the input byte into the low- and high-nibble, which will get
+            // the row-/column-index in the bit-mask.
+            // The bit-mask-matrix looks like
+            //                                     high-nibble
+            // low-nibble  0   1   2   3   4   5   6   7   8   9   A   B   C   D   E   F
+            //         0   1   1   0   0   0   0   1   0   1   1   1   1   1   1   1   1
+            //         1   1   1   0   0   0   0   0   0   1   1   1   1   1   1   1   1
+            //         2   1   1   1   0   0   0   0   0   1   1   1   1   1   1   1   1
+            //         3   1   1   0   0   0   0   0   0   1   1   1   1   1   1   1   1
+            //         4   1   1   0   0   0   0   0   0   1   1   1   1   1   1   1   1
+            //         5   1   1   0   0   0   0   0   0   1   1   1   1   1   1   1   1
+            //         6   1   1   1   0   0   0   0   0   1   1   1   1   1   1   1   1
+            //         7   1   1   1   0   0   0   0   0   1   1   1   1   1   1   1   1
+            //         8   1   1   0   0   0   0   0   0   1   1   1   1   1   1   1   1
+            //         9   1   1   0   0   0   0   0   0   1   1   1   1   1   1   1   1
+            //         A   1   1   0   0   0   0   0   0   1   1   1   1   1   1   1   1
+            //         B   1   1   1   0   0   0   0   0   1   1   1   1   1   1   1   1
+            //         C   1   1   0   1   0   1   0   0   1   1   1   1   1   1   1   1
+            //         D   1   1   0   0   0   0   0   0   1   1   1   1   1   1   1   1
+            //         E   1   1   0   1   0   0   0   0   1   1   1   1   1   1   1   1
+            //         F   1   1   0   0   0   0   0   1   1   1   1   1   1   1   1   1
+            //
+            // where 1 denotes the neeed for escaping, while 0 means no escaping needed.
+            // For high-nibbles in the range 8..F every input needs to be escaped, so we
+            // can omit them in the bit-mask, thus only high-nibbles in the range 0..7 need
+            // to be considered, hence the entries in the bit-mask can be of type byte.
+            //
+            // In the Bitmask (see above) for each row (= low-nibble) a bit-mask for the
+            // high-nibbles (= columns) is created.
+
+            Debug.Assert(Ssse3.IsSupported);
+
+            Vector128<sbyte> highNibbles = Sse2.And(Sse2.ShiftRightLogical(sourceValue.AsInt32(), 4).AsSByte(), s_nibbleMaskSByte);
+            Vector128<sbyte> lowNibbles = Sse2.And(sourceValue, s_nibbleMaskSByte);
+
+            Vector128<sbyte> bitMask = Ssse3.Shuffle(s_bitMask, lowNibbles);
+            Vector128<sbyte> bitPositions = Ssse3.Shuffle(s_bitPosLookup, highNibbles);
+
+            Vector128<sbyte> mask = Sse2.And(bitPositions, bitMask);
+
+            Vector128<sbyte> ne0Gt = Sse2.CompareGreaterThan(mask, s_nullMaskSByte);
```

Comment:
If you're looking for the equivalent of "not equal to zero", I believe the following will produce more efficient codegen:

```cs
return Sse2.CompareEqual(Sse2.CompareEqual(mask, default), default);
```

(And it'd allow getting rid of the static field.)

Reply:
Nice! 👍 
It safes the oring of the tmp-comparisons.

So far I didn't know that the JIT treats `default` here the same way as `Vector128<sbyte>.Zero`.

> (And it'd allow getting rid of the static field.)

The static field is intentional, as the JIT hoists it outside the loop.

<details>
  <summary>w/o static ro field</summary>

```asm
; ...
       C4E25900DB           vpshufb  xmm3, xmm4, xmm3
       C4E25100E7           vpshufb  xmm4, xmm5, xmm7
       C5D9DBDB             vpand    xmm3, xmm4, xmm3
       C5D857E4             vxorps   xmm4, xmm4
       C5D057ED             vxorps   xmm5, xmm5
       C5D174DB             vpcmpeqb xmm3, xmm5, xmm3
       C5D974DB             vpcmpeqb xmm3, xmm4, xmm3
       C5F9D7C3             vpmovmskb eax, xmm3
; ...
```
</details>

<details>
  <summary>with static ro field</summary>

```asm
; ...
       C5F9105808           vmovupd  xmm3, xmmword ptr [rax+8]

G_M60692_IG11:      ;; bbWeight=4   ; loop start
; ...
       C4E25100E4           vpshufb  xmm4, xmm5, xmm4
       C4C24900E9           vpshufb  xmm5, xmm6, xmm9
       C5D1DBE4             vpand    xmm4, xmm5, xmm4
       C5B974E4             vpcmpeqb xmm4, xmm8, xmm4
       C5B974E4             vpcmpeqb xmm4, xmm8, xmm4
       C5F9D7C4             vpmovmskb eax, xmm4
; ...
```
</details>